File: mobile/src/api/feedback.ts
/**
 * src/api/feedback.ts
 * -------------------
 * API calls for the Feedback (game suggestion) endpoint.
 */
import client from './client';

export interface FeedbackRead {
  id: number;
  game_title: string;
  submitted_at: string; // ISO 8601
}

export async function submitFeedback(gameTitle: string): Promise<FeedbackRead> {
  const { data } = await client.post<FeedbackRead>('/feedback/', { game_title: gameTitle });
  return data;
}

File: backend/routers/feedback.py
"""
routers/feedback.py
-------------------
Endpoints for user-submitted game / app suggestions.
"""
from __future__ import annotations
from datetime import datetime
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from database import get_db
from models import Feedback
from schemas import FeedbackCreate, FeedbackRead

router = APIRouter(prefix="/feedback", tags=["Feedback"])


@router.post("/", response_model=FeedbackRead, status_code=201)
def submit_feedback(payload: FeedbackCreate, db: Session = Depends(get_db)):
    """Submit a request for a new game or app."""
    entry = Feedback(
        game_title=payload.game_title.strip(),
        submitted_at=datetime.utcnow(),
    )
    db.add(entry)
    db.commit()
    db.refresh(entry)
    return entry


@router.get("/", response_model=list[FeedbackRead])
def list_feedback(db: Session = Depends(get_db)):
    """Return all submitted feedback entries (admin use)."""
    return db.query(Feedback).order_by(Feedback.submitted_at.desc()).all()
